recipe/meta.yaml

```diff
@@ -13,43 +14,243 @@ source:
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python >=3.6
+      run:
+        - python >=3.6
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+        - boto3
+    test:
+      imports:
+        - parsl
+      commands:
+        - pip check
+      requires:
+        - pip
+
+  - name: {{ name }}-with-kubernetes
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python >=3.6
+      run:
+        - python >=3.6
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+        - kubernetes
+    test:
+      imports:
+        - parsl
+      commands:
+        - pip check
+      requires:
+        - pip
+
+# oauth-ssh not on conda-forge
+#  - name: {{ name }}-with-oauth_ssh
+#    build:
+#      noarch: python
+#    requirements:
+#      host:
+#        - python >=3.6
+#      run:
+#        - python >=3.6
+#        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+#        - oauth-ssh >=0.9
+#    test:
+#      imports:
+#        - parsl
+#      commands:
+#        - pip check
+#      requires:
+#        - pip
+
+  - name: {{ name }}-with-extreme_scale
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python >=3.6
+      run:
+        - python >=3.6
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+        - mpi4py
+    test:
+      imports:
+        - parsl
+      commands:
+        - pip check
+      requires:
+        - pip
+
+  - name: {{ name }}-with-google_cloud
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python >=3.6
+      run:
+        - python >=3.6
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+        - google-auth
+        - google-api-python-client
+    test:
+      imports:
+        - parsl
+      commands:
+        - pip check
+      requires:
+        - pip
+
+  - name: {{ name }}-with-gssapi
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python >=3.6
+      run:
+        - python >=3.6
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+        - python-gssapi
+    test:
+      imports:
+        - parsl
+      commands:
+        - pip check
+      requires:
+        - pip
+
+# azure 3.0.0 was not built for linux
+#  - name: {{ name }}-with-azure
+#    build:
+#      noarch: python
+#    requirements:
+#      host:
+#        - python >=3.6
+#      run:
+#        - python >=3.6
+#        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+#        - azure <=4
+#        - msrestazure
+#    test:
+#      imports:
+#        - parsl
+#      commands:
+#        - pip check
+#      requires:
+#        - pip
+
+  - name: {{ name }}-with-workqueue
+    build:
+      noarch: python
+    requirements:
+      host:
+        - python >=3.6
+      run:
+        - python >=3.6
+        - {{ pin_subpackage(name, max_pin='x.x.x.x.x.x') }}
+        - ndcctools
+        # there is no work_queue pypi or conda-forge package
+        # - work_queue
```

Comment:
I'm following the recommendation from this note: https://github.com/Parsl/parsl/issues/1158#issuecomment-623658664